app/services/schools/archive_coach_service.rb

```diff
@@ -0,0 +1,20 @@
+module Schools
+  class ArchiveCoachService
+    def initialize(faculty, archive)
+      @faculty = faculty
+      @should_archive = archive == "true" ? true : false
+    end
+
+    def execute
+      if !@faculty.exited? && @should_archive
+        @faculty.exited = true
+        @faculty.save!
+        FacultyCohortEnrollment.where(faculty: @faculty).destroy_all
+        FacultyFounderEnrollment.where(faculty: @faculty).destroy_all
+      elsif @faculty.exited? && !@should_archive
+        @faculty.exited = false # unarchive
+        @faculty.save!
+      end
+    end
+  end
+end
```

Comment:
We are not using this service anywhere else, so it doesn't have any impact on extracting it out;